### ISSUE TITLE
Build on Java 12

### ISSUE THREAD

**Issue by dwijnand** [playframework/playframework#9586]
From https://github.com/scala/community-builds/issues/957 it looks like building Play on Java 12 throws:

```
[error] Source option 6 is no longer supported. Use 7 or later.
[error] Target option 6 is no longer supported. Use 7 or later.
[error] (Play-Exceptions / Compile / compileIncremental) javac returned non-zero exit code
```

**Comment by SethTisue** [comment]
this is easily reproducible outside dbuild, and on JDK 13 as well:

```
sbt:Play-Framework> ++2.13.1!
[info] Forcing Scala version to 2.13.1 on all projects.
[info] Reapplying settings...
[info] Set current project to Play-Framework (in build file:/Users/tisue/playframework/)
sbt:Play-Framework> Play-Exceptions/compile
[info] Compiling 2 Java sources to /Users/tisue/playframework/core/play-exceptions/target/classes ...
[error] Source option 6 is no longer supported. Use 7 or later.
[error] Target option 6 is no longer supported. Use 7 or later.
[error] (Play-Exceptions / Compile / compileIncremental) javac returned non-zero exit code
[error] Total time: 0 s, completed Oct 12, 2019, 9:14:18 AM
```

not sure if https://github.com/playframework/interplay/pull/94 might do the trick, or what

**Comment by mkurz** [comment]
This got fixed by https://github.com/playframework/interplay/pull/101 which dropped support for sbt 0.13 in interplay, which allowed us to get rid of Scala 2.10 and therefore also any `-source` and `-target` flag with `1.6` got removed: https://github.com/playframework/interplay/pull/101/files#diff-9f23263fad304b1ca48c64c38db7d8f20fbd47120c76899715b9913021ab96eeL26
So for Play's `master` branch this issue can be closed (because on master we already use interplay 3.0 which includes the referenced pull request)

I am aware that the jdk 16 community build however builds against Play's 2.8.x branch, therefore I tried to fix this in interplay's 2.1.x branch as well via https://github.com/playframework/interplay/pull/94 - however I was out of luck.

This made me realize that I don't think it makes sense to invest time to make Play 2.8.x work with jdk 16 and upcoming 17 LTS because I don't think that will be successful. The 2.8.x branch will never be fully compatible with jdk 16/17, already because Guice 4.2.x (which depends on ASM) won't work with the newer JDK (Guice 5 however will be released soon with JDK 16/17 support, however I don't think it's a good idea to make such a major version bump in 2.8.x).

My personal conclusion from all of this is that we need to cut Play 2.9 sometimes this year which will then support the latest JDK's (and probably also Scala 3)
